app/src/main/java/com/example/tasky/feature_agenda/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -2,114 +2,161 @@ package com.example.tasky.feature_agenda.data.repository
-                return@flow
-            }
+        } catch (e: HttpException) {
+            return AgendaResult.Error(e.message ?: "Invalid Response")
+        } catch (e: IOException) {
+            return AgendaResult.Error(e.message ?: "Couldn't reach server")
+        }
+
+        remoteAgenda.let { agenda ->
+            db.eventDao.deleteEvents()
+            db.eventDao.deleteAttendees()
+            db.reminderDao.deleteReminders()
+            db.taskDao.deleteTasks()
 
-            val remoteAgenda = try {
-                val response = api.getAgenda(
-                    timeZone = TimeZone.getDefault().toString(),
-                    time = System.currentTimeMillis()
-                )
+            val eventList = agenda.filterIsInstance<AgendaItem.Event>()
+            val remindersList = agenda.filterIsInstance<AgendaItem.Reminder>()
+            val tasksList = agenda.filterIsInstance<AgendaItem.Task>()
 
-                val remoteEvents = response.body()?.events?.map { it.toEvent() } ?: emptyList()
-                val remoteTasks = response.body()?.tasks?.map { it.toTask() } ?: emptyList()
-                val remoteReminders = response.body()?.reminders?.map { it.toReminder() } ?: emptyList()
+            eventList.forEach { event->
+                db.eventDao.upsertEvent(event.toEventEntity())
+                db.eventDao.upsertAttendee(event.attendees.map { it.toAttendeeEntity(event.eventId) })
+            }
 
-                remoteEvents + remoteTasks + remoteReminders
+            tasksList.forEach {
+                db.taskDao.upsertTask(it.toTaskEntity())
+            }
 
-            } catch (e: HttpException) {
-                emit(Resource.Error(e.message ?: "Couldn't load data"))
-                null
-            } catch (e: IOException) {
-                emit(Resource.Error(e.message ?: "Couldn't load data"))
-                null
+            remindersList.forEach {
+                db.reminderDao.upsertReminder(it.toReminderEntity())
             }
 
-            remoteAgenda?.let { agenda ->
-                daos.eventDao.deleteEvents()
-                daos.eventDao.deleteAttendees()
-                daos.taskDao.deleteTasks()
-                daos.reminderDao.deleteReminders()
+            return AgendaResult.Success
+        }
+    }
+
+    override suspend fun syncServerWithLocallyDeletedItems(
+        eventIds: List<String>,
+        reminderIds: List<String>,
+        taskIds: List<String>
+    ): AgendaResult {
+        return try {
+            api.syncAgenda(SyncAgendaRequest(
+                deletedEventIds = eventIds,
+                deletedReminderIds = reminderIds,
+                deletedTaskIds = taskIds
+            ))
+            AgendaResult.Success
+        } catch(e: HttpException) {
+            AgendaResult.Error(e.message ?: "Invalid Response")
+        } catch(e: IOException) {
+            AgendaResult.Error(e.message ?: "Couldn't reach server")
+        }
+    }
```

Comment:
I'd personally expose a single `syncAgenda` function which performs everything necessary in the right order. The implementation of the caching strategy is typically also part of the repository, so you wouldn't need to expose this `syncServerWithLocallyDeletedItems` function (you can still keep it as a private one)